### PR TITLE
feat: add "rejection reason" to applications and include in email ❎

### LINKED ISSUE #216: Add context/messaging to frequent rejection reasons 💬
### Description

We get a lot of emails from applicants on why they were rejected from the community. To fix this, we should add some more context to the rejection email. This should ideally result in the less emails being sent to our support.

Here are some frequent rejection reasons:
- Not an undergrad student.
- Incorrect or suspicious LinkedIn.
- Not the right major.
- Not enrolled in US or Canada.

### Acceptance Criteria

For this issue to be complete, you'll need to:
- Make the "Reject" button go to a new modal that displays a form that allows you to select a "Reason".
- Show the "Rejection Reason" in the "Applications" table in the Admin Dashboard. Note that this should only be in the "Rejected" and "All" views.
- Update the email template `application-rejected` email template to take in a dynamic `reason` and display it accordingly.

### PR DESCRIPTION
**Pull request by arelymartinez16**
## Description ✏️

Closes #216 

Key Features:

- Created a modal that shows when someone rejects an application and they can provide a reason why they're rejecting the application before rejecting
- Created "rejectionReason" column under the "applications" table in the database
- Rejection reason is shown on the UI in the "Rejected" and "All" views
- Include the rejection reason in the email template 

## Type of Change 🐞

- [x] Feature - A non-breaking change which adds functionality.
- [ ] Fix - A non-breaking change which fixes an issue.
- [ ] Refactor - A change that neither fixes a bug nor adds a feature.
- [ ] Documentation - A change only to in-code or markdown documentation.
- [ ] Tests - A change that adds missing unit/integration tests.
- [ ] Chore - A change that is likely none of the above.

## Checklist ✅

- [X] I have done a self-review of my code.
- [ ] I have manually tested my code (if applicable).
- [ ] I have added/updated any relevant documentation (if applicable).

### CODE REVIEW

File: apps/admin-dashboard/app/routes/_dashboard.applications.tsx [hunk 1]
```diff
@@ -197,6 +199,15 @@ function ApplicationsTable() {
         return `${reviewedByFirstName} ${reviewedByLastName}`;
       },
     },
+    // TODO: Show the "Rejection Reason" in the "Applications" table in the Admin Dashboard. Note that this should only be in the "Rejected" and "All" views.
+    // Have to add a column of the rejection reason in the database
```
Comment (tomas-salgado) [inline apps/admin-dashboard/app/routes/_dashboard.applications.tsx:203]: Let's remove these comments-looks like this is all done!

File: packages/core/src/modules/application/application.core.ts [hunk 2]
```diff
@@ -368,6 +370,8 @@ export async function rejectApplication(
         rejectedAt: new Date(),
         reviewedById: adminId,
         status: ApplicationStatus.REJECTED,
+        // rejectionReason: reason || null,
```
Comment (tomas-salgado) [inline packages/core/src/modules/application/application.core.ts:373]: remove this comment

**Review by tomas-salgado — CHANGES_REQUESTED** [review]
This is amazing work @arelymartinez16!! There were a lot of different parts to this issue but you handled it all really well-just tested it locally and it's working great! I left a few minor comments, but this should soon be good to go 🚀

File: packages/db/src/migrations/20240819195536_add_rejection_reason_to_applications_table.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { type Kysely } from 'kysely';
```
Comment (tomas-salgado) [inline packages/db/src/migrations/20240819195536_add_rejection_reason_to_applications_table.ts:1]: Note for @ramiAbdou: reminder to update the timestamp in file name

**Review by tomas-salgado — APPROVED** [review]
LGTM! Rami will take a final look and we'll ship it 🚢

**Review by ramiAbdou — APPROVED** [review]
Thanks for all the work here, @arelymartinez16! I made a couple tweaks to the UX that weren't in the original issue (mainly using a dropdown instead of a modal and some updated email copy).

I can't tell you how excited Jennifer (our community manager) is for these changes because we get so many emails asking why an application was rejected 😭

The impact of this will be huge! 🚀

File: packages/email-templates/emails/application-rejected.tsx [hunk 4]
```diff
@@ -22,6 +23,8 @@ export function ApplicationRejectedEmail({
           into our community at this time.
         </Email.Text>
 
+        <Email.Text>Rejection Reason: {rejectionReason}</Email.Text>
+
```
[the remark this answers is not in the document]
Comment (ramiAbdou) [reply packages/email-templates/emails/application-rejected.tsx]: Updated!

**Pull request merged**